### DISCUSSION TITLE
Blas/ lapack/boost/eigen

### DISCUSSION THREAD

**Discussion by samukweku** [h2oai/datatable#3301] [General] [answered]
Does datatable use blas/lapack? If it doesn't is there any reason not to? I assume it would offer more performance if used

**Comment by oleksiyskononenko** [comment, marked as answer]
Nope, all the datatable dependencies are listed here: https://github.com/h2oai/datatable/blob/main/ci/ext.py#L418-L426
We also have some borrowed code with the appropriate licenses here: https://github.com/h2oai/datatable/tree/main/src/core/lib

As for the blas/lapack/boost — do you have any specific routines in mind to be re-used from these libraries? These are pretty big libs, adding them as dependency will hugely increase amount of the datatable source code and also the building time.

**Reply by samukweku** [reply]
just an assumption, based on its usage within numpy. I just assumed, with no proofs, that it gives some performance boost to warrant the dependencies.

**Reply by oleksiyskononenko** [reply]
I guess numpy is using lapack/blas for linear algebra routines: https://numpy.org/doc/stable/reference/routines.linalg.html In datatable I don't think we have any linear algebra tools. We only have `dt.models.LinearModel`, but that's another story.
